src/imitation/util/logger.py

```diff
@@ -0,0 +1,200 @@
+import contextlib
+import os
+from typing import ContextManager, Optional, Sequence
+
+import stable_baselines.logger as sb_logger
+
+
+def _build_output_formats(folder: str,
+                          format_strs: Sequence[str] = None,
+                          ) -> Sequence[sb_logger.KVWriter]:
+  """Build output formats for initializing a Stable Baselines Logger.
+
+  Args:
+    folder: Path to directory that logs are written to.
+    format_strs: An list of output format strings. For details on available
+      output formats see `stable_baselines.logger.make_output_format`.
+  """
+  os.makedirs(folder, exist_ok=True)
+  output_formats = [sb_logger.make_output_format(f, folder)
+                    for f in format_strs]
+  return output_formats
+
+
+class _AccumulatingLogger(sb_logger.Logger):
+
+  def __init__(self,
+               format_strs: Sequence[str],
+               *,
+               mean_logger: sb_logger.Logger,
+               subdir: str):
+    """Like Logger, except also accumulates logkv_mean on the mean_logger.
+
+    Args:
+      format_strs: An list of output format strings. For details on available
+        output formats see `stable_baselines.logger.make_output_format`.
+      mean_logger: A background logger that is used to keep track of log
+        means.
+      subdir: Used to build the logging directory. Also used in the logging
+        prefix for every key written to both this logger and `mean_logger`.
+    """
+    self.subdir = subdir
+    self.mean_logger = mean_logger
+    folder = os.path.join(self.mean_logger.dir, "raw", subdir)
```

Comment:
Should mention in the docstring that we write output in subdirectories of `mean_logger`

Reply:
done